packages/api/cms-api/src/generator/generate-crud-input.ts

```diff
@@ -103,11 +108,31 @@ export async function generateCrudInput(generatorOptions: { targetDirectory: str
             decorators.push("@IsUUID()");
             type = "string";
         } else if (prop.reference == "1:m") {
-            decorators.length = 0;
-            decorators.push(`@Field(() => [ID])`);
-            decorators.push(`@IsArray()`);
-            decorators.push(`@IsUUID(undefined, { each: true })`);
-            type = "string[]";
+            if (prop.orphanRemoval) {
+                //if orphanRemoval is enabled, we need to generate a nested input type
```

Comment:
I don't understand why the nested input type depends on orphanRemoval.

Reply:
There are to kinds of 1:n relation apis that get generated:

- input contains ids (useful for pools, good example is a Product Tag or the reverse relation of Product Categories)
- input contains complete relation data (useful when data exclusively belongs to parent, good example are Product Variants)

It makes absolutely sense do enable orphanRemoval for (2) and not for (1), so I decided I'll re-use this property. Alternative would be a new decorator, but I thought the less custom decorators we need the better.

Maybe this is too much magic.